internal/liteconfig/config.go

```diff
@@ -175,12 +177,24 @@ func (o *Config) mustGetService(frontendPortOffset int) config.Service {
 		}
 		membershipPort = o.portProvider.mustGetFreePort()
 	}
-	return config.Service{
-		RPC: config.RPC{
-			GRPCPort:        grpcPort,
-			MembershipPort:  membershipPort,
-			BindOnLocalHost: true,
-			BindOnIP:        "",
-		},
+
+	if grpcPort == o.FrontendPort && o.FrontendAddress != broadcastAddress {
+		return config.Service{
+			RPC: config.RPC{
+				GRPCPort:        grpcPort,
+				MembershipPort:  membershipPort,
+				BindOnLocalHost: false,
+				BindOnIP:        o.FrontendAddress,
+			},
+		}
+	} else {
+		return config.Service{
+			RPC: config.RPC{
+				GRPCPort:        grpcPort,
+				MembershipPort:  membershipPort,
+				BindOnLocalHost: true,
+				BindOnIP:        "",
+			},
+		}
```

Comment:
Nit: I think this whole function would be a little more clear if we refactored to initialize a service config and then conditionally modify its fields before returning. We could also add some code comments. Eg.

```go
func (o *Config) mustGetService(frontendPortOffset int) config.Service {
	svc := config.Service{
		RPC: config.RPC{
			GRPCPort:        o.FrontendPort + frontendPortOffset
			MembershipPort:  o.FrontendPort + 100 + frontendPortOffset
			BindOnLocalHost: true,
			BindOnIP:        "",
		}
	}

	// Assign any open port when configured to use dynamic ports
	if o.DynamicPorts {
		if frontendPortOffset != 0 {
			svc.RPC.GRPCPort = o.portProvider.mustGetFreePort()
		}
		svc.RPC.MembershipPort = o.portProvider.mustGetFreePort()
	}

	// Optionally bind frontend to IPv4 address
	if frontendPortOffset == 0 && o.FrontendAddress != "" {
		svc.RPC.BindOnLocalHost = false
		svc.RPC.BindOnIP = o.FrontendAddress
	}

	return svc
}

```

Reply:
Moved this and pulled in changes here
https://github.com/DataDog/temporalite/pull/17/files